tests/spec/enable_http_compression.rb

```diff
@@ -0,0 +1,27 @@
+require 'net/http'
+
+feature "Change http compression" do
+	after(:all) do
+                set_config "apache.compression": "false"
+                wait_for_nextcloud
+	end
+
+	scenario "compression" do
+                set_config "apache.compression": "true"
+                wait_for_nextcloud
+
+		assert_apache_compression
+	end
+
+	protected
+
+	def assert_apache_compression
+		# Verify that Apache is returning compressed reponses
+		expect(nextcloud_response["content-encoding"]).to eq "br"
```

Comment:
Lovely tests my friend, good work! Can you add one more that confirms that it's NOT compressed by default? Also check your indentation, looks like a mixture of tabs and spaces in here.

Reply:
Sure, I will add it when I can. Just one question: how can I run the tests?

Reply:
You should be able to [follow this](https://github.com/nextcloud/nextcloud-snap#hacking).

Reply:
Well, I'm trying to run this but I'm getting errors. bundle install seems to be OK:
```
$ cd tests/ && bundle install
Using rake 13.0.1
Using public_suffix 4.0.5
Using addressable 2.7.0
Using bundler 2.1.4
Using mini_mime 1.0.2
Using mini_portile2 2.4.0
Using nokogiri 1.10.9
Using rack 2.2.3
Using rack-test 1.1.0
Using regexp_parser 1.7.1
Using xpath 3.2.0
Using capybara 3.33.0
Using childprocess 3.0.0
Using diff-lcs 1.4.2
Using rspec-support 3.9.3
Using rspec-core 3.9.2
Using rspec-expectations 3.9.2
Using rspec-mocks 3.9.1
Using rspec 3.9.0
Using rubyzip 2.3.0
Using selenium-webdriver 3.142.7
Bundle complete! 4 Gemfile dependencies, 21 gems now installed.
Bundled gems are installed into `./vendor/bundle`
```
But then:
```
$ rake test
rake aborted!
LoadError: cannot load such file -- rspec/core/rake_task
/home/marc/Software/github/pachulo/nextcloud-snap/tests/Rakefile:2:in `<top (required)>'
/usr/share/rubygems-integration/all/gems/rake-13.0.1/exe/rake:27:in `<top (required)>'
(See full trace by running task with --trace)
```
What am I doing wrong?

Reply:
Hmm, what version of ruby are you using? I recommend installing [RVM](https://rvm.io/), then the .ruby-version and .ruby-gemset will ensure we're using similar configurations.

Reply:
> I recommend installing RVM, then the .ruby-version and .ruby-gemset will ensure we're using similar configurations.

Shall we update the instructions here then? https://github.com/nextcloud-snap/nextcloud-snap#hacking

Reply:
OK, after using RVM I was able to install Ruby and the gems to execute the tests.
Now I'm facing the problem that the `compression` scenario does not work: I'm not really sure why I'm getting a `nil` instead of the HTTP response...do you have any clue @kyrofa ?

```
$ /home/marc/.rvm/rubies/ruby-2.6.2/bin/ruby -I/home/marc/.rvm/gems/ruby-2.6.2@nextcloud-test/gems/rspec-core-3.10.1/lib:/home/marc/.rvm/gems/ruby-2.6.2@nextcloud-test/gems/rspec-support-3.10.2/lib /home/marc/.rvm/gems/ruby-2.6.2@nextcloud-test/gems/rspec-core-3.10.1/exe/rspec integration/enable_http_compression.rb --format documentation

Change http compression
  invalid
  compression (FAILED - 1)

Failures:

  1) Change http compression compression
     Failure/Error: expect(response["content-encoding"]).to include "br"
       expected nil to include "br", but it does not respond to `include?`
     # ./integration/enable_http_compression.rb:28:in `assert_apache_compression'
     # ./integration/enable_http_compression.rb:20:in `block (2 levels) in <top (required)>'

Finished in 14.41 seconds (files took 0.42254 seconds to load)
2 examples, 1 failure

Failed examples:

rspec ./integration/enable_http_compression.rb:16 # Change http compression compression
```

Reply:
The test looks sane to me. You seem to be waiting properly. Try adding a `puts response.inspect` before the `expect`, what does the response look like?

Reply:
`puts response.to_hash` might be more helpful.

Reply:
I think I solved this: I needed to set up the `Accept-Encoding` HTTP header to get what I wanted.

Reply:
> Can you add one more that confirms that it's NOT compressed by default?

I've added it as part of the `compression` scenario, is that OK?